### PR TITLE
Add a Saml2MetadataFilter implementation that can be used to return SAML metadata to a caller.

### PR DESCRIPTION
**Pull request by minfrin**

### CODE REVIEW

File: pac4j-saml/src/main/java/org/pac4j/saml/metadata/Saml2MetadataFilter.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * Example shiro.ini configuration:
+ * 
+ * saml2MetadataFilter = org.pac4j.saml.metadata.Saml2MetadataFilter
+ * saml2MetadataFilter.config = $config
+ * 
+ * [urls]
+ * /API/SAML2/metadata = saml2MetadataFilter
+ * 
+ * @author Graham Leggett
+ */
```
[the remark this answers is not in the document]
Comment (leleuj) [reply pac4j-saml/src/main/java/org/pac4j/saml/metadata/Saml2MetadataFilter.java]: It should be 3.8.0.

[the remark this answers is not in the document]
Comment (minfrin) [reply pac4j-saml/src/main/java/org/pac4j/saml/metadata/Saml2MetadataFilter.java]: It's done.

File: pac4j-saml/src/main/java/org/pac4j/saml/metadata/Saml2MetadataFilter.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    public void init(final FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public void doFilter(final ServletRequest servletRequest, final ServletResponse servletResponse,
+            final FilterChain filterChain) throws IOException, ServletException {
+
+        CommonHelper.assertNotNull("config", config);
+
+        SAML2Client client = (SAML2Client) config.getClients().findClient("SAML2Client").get();
```
[the remark this answers is not in the document]
Comment (minfrin) [reply pac4j-saml/src/main/java/org/pac4j/saml/metadata/Saml2MetadataFilter.java]: True - not sure how I missed that! Done.

**Comment by minfrin** [comment]
Can you confirm this is ok? As soon as it's done, I'll re-fork and sort out v3.8.0.

**Comment by minfrin** [comment]
Reforked, backport for 3.8.0 proposed in https://github.com/pac4j/pac4j/pull/1314.

File: pac4j-saml/src/main/java/org/pac4j/saml/metadata/Saml2MetadataFilter.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ * /API/SAML2/metadata = saml2MetadataFilter
+ * 
+ * @author Graham Leggett
+ * @since 3.8.0
+ */
+public class Saml2MetadataFilter implements Filter {
+
+    private Config config;
+
+    private String client;
```
Comment (leleuj) [inline pac4j-saml/src/main/java/org/pac4j/saml/metadata/Saml2MetadataFilter.java]: I'd prefer the property to be named `clientName`.

Comment (minfrin) [reply pac4j-saml/src/main/java/org/pac4j/saml/metadata/Saml2MetadataFilter.java]: It's done, made a new PR https://github.com/pac4j/pac4j/pull/1315.

**Comment by minfrin** [comment]
Replaced with https://github.com/pac4j/pac4j/pull/1315.

**Pull request closed without merging**